suite/tests/security-common/selfmod.c

```diff
@@ -428,22 +555,61 @@ ADDRTAKEN_LABEL(last_byte_jmp_no_ilt:)
         jmp      last_byte_ret_zero     /* 1 byte opcode + 1 byte rel offset */
 last_byte_ret_one:
         mov      eax, HEX(1)
+#else
+        nop
+DECLARE_GLOBAL(last_byte_jmp_label)
+ADDRTAKEN_LABEL(last_byte_jmp_label:)
+        b last_byte_jmp
+last_byte_ret_zero:
+        mov x0, #0
+        ret
+last_byte_jmp:
+        b last_byte_ret_zero
+last_byte_ret_one:
+        mov x0, #1
+#endif
         ret
         END_FUNC(FUNCNAME)
 #undef FUNCNAME
 
 ALIGN_WITH_NOPS(16)
 
+#ifdef X86
 #define FUNCNAME make_last_byte_selfmod
         DECLARE_FUNC(FUNCNAME)
 GLOBAL_LABEL(FUNCNAME:)
+#ifdef X86
         lea      REG_XAX, SYMREF(last_byte_immed_plus_four - 4)
         mov      DWORD [REG_XAX], HEX(0)     /* selfmod write */
         mov      REG_XAX, HEX(0)             /* mov_imm to modify */
 ADDRTAKEN_LABEL(last_byte_immed_plus_four:)
+#else
+        adr x9, last_instruction - 4
+        ldr w10, [x9]
+        mov w11, #1
```

Comment:
What is this?  Is this the instr that gets modified?  Why `#1`?  Add a comment?

Reply:
This code was left in by accident and is never built. Will remove.